### ISSUE TITLE
XML Parsing Error Prevents Pages from Loading

### ISSUE THREAD

**Issue by mathmaniac43** [mozilla-mobile/fenix#19580]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce

Attempt to load the following webpage (some others fail too, but I have not found a specific pattern yet).
https://support.mozilla.org/1/mobile/89.0.0-beta.8/Android/en-US/faq-android

### Expected behavior

The webpage loads.

### Actual behavior

An XML parsing error related to Gecko appears.
![Screenshot_20210520-060132.png](https://user-images.githubusercontent.com/918497/118960915-c388f180-b931-11eb-8465-e8f47ef068af.png)

### Device information

* Device vendor / model and Android version:

Motorola Moto X4, Android 9

* Firefox for Android version: ? (go to Settings -> About Firefox)
 
Latest Beta
![Screenshot_20210520-060103.png](https://user-images.githubusercontent.com/918497/118961037-e1eeed00-b931-11eb-8ffe-5698467eb3c2.png)

**Comment by SoftVision-LorandJanos** [comment]
Hi @mathmaniac43 ! Thank you for your report!
I tried to reproduce the issue described by you, many times, but I was not able to, using the latest Beta (89.0.0-beta.8). 
Could you help us with a video about the issue? Also can you provide with some additional sites where you had the same issue? Maybe try reproducing it on the latest Nightly. Thank you!
Device used: Samsung Galaxy Tab 3 with Android 9.

**Comment by mathmaniac43** [comment]
Hello! I would be happy to try and help at the end of my workday. How do you recommend I record the video, where should I upload, and what would you like to see? Just me visiting a few sites that exhibit the problem? Any debug information I can try to show while I drive?

I did test this with the same Firefox Beta on my Samsung Galaxy Tab A but was unable to reproduce on that. I have a few other Android devices lying around, if I have time I can see if they do the same thing or not. I only have one Motorola though.

Thanks!

**Comment by SoftVision-LorandJanos** [comment]
Hi @mathmaniac43 ! Well, any tool that permits you screen recording is fine.
Please upload it/them here as a comment.
Yes, visiting a few sites that exhibit the problem would help a lot.
Debug information could help the one(s) who will work on this issue, so if it's not too much, you can attach any information you can provide with.
Also, yes please check if you can reproduce on other devices as well, whenever you have time for it, and maybe try on the same device you mentioned in this issue but with the latest Nightly or clear the data on the Beta you tested on to see if it's not influenced by anything else.
Thank you!

**Comment by mathmaniac43** [comment]
Very strange. When I just went to test it, NO websites would load. I cleared cache, no luck; I did a full reinstall, and best I can tell, no issue anymore. I think I just got into some kind of funky state. I think we should close this unless it comes back for me or other users...

**Comment by Halmex** [comment]
Hi, I just got the same error here a  [video](https://user-images.githubusercontent.com/13755401/119238776-f6570380-bb44-11eb-8d54-df9dce3c7870.mp4), basically no website works. I have no clue about the debugging tools, maybe this helps anyway. I saw nothing in the console of the main process, so I recorded opening 2 websites with the profiler. Once with the Web Developer and once with Firefox Platform option. [https://share.firefox.dev/2RGWBuX](https://share.firefox.dev/2RGWBuX), [https://share.firefox.dev/3vc3u5Z](https://share.firefox.dev/3vc3u5Z)

**Comment by botistreamer** [comment]
I am also facing the same issue on Poco X3, Android 11, ArrowOS on all websites.
This happened suddenly today and no sites seems to open now. No idea what triggered this.
Firefox 89.1.1 (Build #2015812947)


https://user-images.githubusercontent.com/47123395/123551467-87964700-d78f-11eb-9bae-652114cfbd50.mp4

**Comment by mduchev** [comment]
This **must** be a top priority. I've also faced this issue. What I've done in order to make it happen is simply opened the History page from the 3-dot menu, scrolled a bit and then tried to open a page from my history. Then I got the same YSOD.

Here's a gif of the issue:
![screen-20210810-214204](https://user-images.githubusercontent.com/8217099/128918252-d24e59c4-303e-4265-8db3-f06767810407.gif)

Firefox is unsuable as soon as this issue appears. Refreshing, manual entering of anither site, closing the all tabs and opening a brand new tab and trying something - nothing works.

Closing and opening the browser again is what fixes the problem.

Phone: Pixel 4a 5G
OS: Android Beta 3
Firefox: 91.1.0
